### ISSUE TITLE
Refresh github webhook doesn't work with branch name using a slash `/`

### ISSUE THREAD

**Issue by jpaille** [argoproj/argo-cd#8005]
**Describe the bug**

We use Argocd application on different branches named as `environment/<team_name>`  for example environment/b2b
The refresh webhook is ignored in argocd-server because of this check `appRevisionHasChanged`
https://github.com/argoproj/argo-cd/blob/master/util/webhook/webhook.go#L238
and this function 
```
func parseRevision(ref string) string {
	refParts := strings.SplitN(ref, "/", 3)
	return refParts[len(refParts)-1]
} 
```
This function removes `/` from the revision so `environment/b2b` become just `b2b`. `appRevisionHasChanged` returns false and the refresh doesn't occur.

**To Reproduce**

- Create an app with target revision a branch name using a slash.
- Try to trigger the github push webhook 

**Version**

We use argocd 2.1.2 but the issue is still present in argocd latest

**Logs**

From argocd-server:
```
time="2021-12-21T15:58:03Z" level=info msg="Received push event repo: https://github.com/XXX/YYYY, revision: environment/b2b, touchedHead: false"
```
The push event is received by argocd-server but ignored (the application is not refreshed)

**Comment by jomenxiao** [comment]
Fixed by #7939

**Comment by jpaille** [comment]
Indeed the issue is already fixed by #7939
Thank you
